nobl9/resource_direct.go

```diff
@@ -35,14 +36,15 @@ func resourceDirectFactory(directSpec directSpecResource) *schema.Resource {
 			"project":      schemaProject(),
 			"description":  schemaDescription(),
 			"source_of": {
+				Deprecated:  "source_of is Deprecated and not used anywhere. You can safely remove it from your configuration file.",
 				Type:        schema.TypeList,
-				Required:    true,
+				Optional:    true,
 				MinItems:    1,
 				MaxItems:    2,
-				Description: "Source of Metrics and/or Services.",
+				Description: "Deprecated value that was used to indicate whether the field was a source of Metrics and/or Services. It's not used anywhere but kept for backward compatibility.",
```

Comment:
nitpick:  It's not used anywhere **but, kept** for backward compatibility.

Reply:
As discussed on Slack - irrelevant after the new changes. Thanks!